storm-core/src/jvm/org/apache/storm/daemon/supervisor/timer/RunProfilerActions.java

```diff
@@ -0,0 +1,214 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.daemon.supervisor.timer;
+
+import com.google.common.collect.Lists;
+import org.apache.storm.Config;
+import org.apache.storm.cluster.IStormClusterState;
+import org.apache.storm.daemon.supervisor.SupervisorData;
+import org.apache.storm.daemon.supervisor.SupervisorUtils;
+import org.apache.storm.generated.ProfileAction;
+import org.apache.storm.generated.ProfileRequest;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class RunProfilerActions implements Runnable {
+    private static Logger LOG = LoggerFactory.getLogger(RunProfilerActions.class);
+
+    private Map conf;
+    private IStormClusterState stormClusterState;
+    private String hostName;
+
+    private String profileCmd;
+
+    private SupervisorData supervisorData;
+
+    private class ActionExitCallback implements Utils.ExitCodeCallable {
+        private String stormId;
+        private ProfileRequest profileRequest;
+        private String logPrefix;
+        private boolean stop;
+
+        public ActionExitCallback(String stormId, ProfileRequest profileRequest, String logPrefix, boolean stop) {
+            this.stormId = stormId;
+            this.profileRequest = profileRequest;
+            this.logPrefix = logPrefix;
+            this.stop = stop;
+        }
+
+        @Override
+        public Object call() throws Exception {
+            return null;
+        }
+
+        @Override
+        public Object call(int exitCode) {
+            LOG.info("{} profile-action exited for {}", logPrefix, exitCode);
+            try {
+                if (stop)
+                    stormClusterState.deleteTopologyProfileRequests(stormId, profileRequest);
+            } catch (Exception e) {
+                LOG.warn("failed delete profileRequest: " + profileRequest);
+            }
+            return null;
+        }
+    }
+
+    public RunProfilerActions(SupervisorData supervisorData) {
+        this.conf = supervisorData.getConf();
+        this.stormClusterState = supervisorData.getStormClusterState();
+        this.hostName = supervisorData.getHostName();
+        this.profileCmd = (String) (conf.get(Config.WORKER_PROFILER_COMMAND));
```

Comment:
The original code had this relative to the `storm.home/bin`

```
            profile-cmd (str (clojure.java.io/file storm-home
                                                   "bin"
                                                   (conf WORKER-PROFILER-COMMAND)))
```
Probably should be
```
       String stormHome = System.getProperty("storm.home");
       this.profileCmd = stormHome + Utils.FILE_PATH_SEPARATOR + "bin" + Utils.FILE_PATH_SEPARATOR + conf.get(Config.WORKER_PROFILER_COMMAND));
```